common/djangoapps/enrollment/views.py

```diff
@@ -515,9 +516,17 @@ def post(self, request):
         go through `add_enrollment()`, which allows creation of new and reactivation of old enrollments.
         """
         # Get the User, Course ID, and Mode from the request.
+        user = request.data.get('user', request.user.username)
+
+        email_pattern = r'^\w+[\.\w\+]*\w+@\w+\.\w{2,3}$'
+
+        if re.match(email_pattern, user):
+            email = user
+            username = request.user.username
```

Comment:
if we explicitly pass user parameter we don't need to fallback to request.user, please remove this assigning

Reply:
A request could be without 'user' parameter. We need username to check who made this call (staff or else), I decided to move this from [this line](https://github.com/raccoongang/edx-platform/pull/2085/files#diff-a4a6f81c03f035ebb4441f342e27c62eL545) to keep vars close and improve readability